pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -236,6 +248,70 @@ func (pc *PackageControllerClient) Enable(ctx context.Context) error {
 	return nil
 }
 
+// UpdateSecrets is used to update the registry-mirror-cred secret used by the packages controller.
+func (pc *PackageControllerClient) UpdateSecrets(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
+	secretName := "registry-mirror-cred"
+	secretKey := types.NamespacedName{
+		Namespace: constants.EksaPackagesName,
+		Name:      secretName,
+	}
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: constants.EksaPackagesName,
+		},
+	}
+	credErr := client.Get(ctx, secretKey, secret)
+	err := fillRegistrySecret(cluster.Name, cluster.Spec.RegistryMirrorConfiguration, secret)
+	if err != nil {
+		return err
+	}
+
+	if apierrors.IsNotFound(credErr) {
```

Comment:
if you just want to do an "upsert", you don't need to do a get. Just do the upgrade and if the object doesn't exist, it will return a IsNotFound. Then you can call create

However, if you do a `Get` first, at least check if the object needs to be updated or not. That way we avoid an extra write call. Remember that this will be called on every reconcile loop.